mhw_library/utils.py

```diff
@@ -1,15 +1,362 @@
 
-def create_events_file(detrended, blobs):
-    '''
-    Merges SST anomalies with the MHW identifier labels.
     Parameters
     ----------
-    detrended : xarray.DataArray (time:1980, lat: 192, lon: 288)
-        Consists of SST anomalies.
-    blobs : xarray.DataArray 'labels' (time: 1980, lat: 192, lon: 288)
-        Consists of MHW identifier labels.
-    '''
+    detrended : xarray.DataArray of shape (time: 1980, lat: 192, lon: 288)
+        A data array containing SST anomalies.
+    blobs : xarray.DataArray of shape (time: 1980, lat: 192, lon: 288)
+        A data array containing object identifier labels.
+
+    Returns
+    -------
+    xarray.Dataset
+        A dataset containing the following data variables:
+        - 'SSTA': SST anomaly data array
+        - 'labels': object identifier labels data array
+    """
     detrended.name = 'SSTA'
-    return xr.merge([detrended, blobs])
\ No newline at end of file
+    return xr.merge([detrended, blobs])
+
+def count_objects(merged_xarray: xr.Dataset) -> int:
+    """
+    Counts the total number of objects in an event file.
+
+    Parameters
+    ----------
+    merged_xarray : xr.Dataset
+        A merged xarray containing the data variables 'SSTA' and 'labels'.
+
+    Returns
+    -------
+    int
+        The total number of objects in the 'labels' data variable, excluding the background.
+
+    Notes
+    -----
+    The 'labels' data variable should contain integer labels that identify connected regions in the 'SSTA' data variable.
+    The value 0 is reserved for the background and is not considered an object.
+    """
+    return len(np.unique(merged_xarray.labels)) - 1
+
+def calc_initialization(merged_xarray: xr.Dataset, mhw_id: int) -> Tuple[int, xr.DataArray, int]:
+    """
+    Get initialization state information for a specific object in a merged xarray.
+
+    Parameters
+    ----------
+    merged_xarray : xarray.Dataset
+        A merged xarray containing the data variables 'SSTA' and 'labels'.
+    mhw_id : int
+        A specific object identifier.
+
+    Returns
+    -------
+    Tuple[int, xr.DataArray, int]
+        A tuple containing the following information about the object:
+        1. The timestep when the object first appears in the 'labels' data variable.
+        2. The image of the object at the initialization timestep, represented as an xarray DataArray object
+           with dimensions ('latitude', 'longitude').
+        3. The month (1-12) when the object initializes.
+
+    Notes
+    -----
+    This function assumes that the first timestep in the model corresponds to the first month of the year.
+    """
+    one_obj = merged_xarray.where(merged_xarray.labels==mhw_id, drop=False)
+    mhw_when = np.argwhere(one_obj.labels.max(axis=(1,2)).values > 0.)
+    first_timestep = mhw_when[0][0]
+    bymonth = np.resize(np.arange(1,13),12*166)[1:-11]
+    month = bymonth[first_timestep]
+    return first_timestep, one_obj.SSTA[first_timestep,:,:], month
```

Comment:
You could order them first_timestep, month, image?  Just so first timestep and month are next to each other since they describe a similar thing.